### ISSUE TITLE
GeneralErrors thrown by Hibernate during database operations

### ISSUE THREAD

**Issue by kiniry** [FreeAndFair/ColoradoRLA#395]
With the permission of CDOS, we grabbed the server log at 16:01 on 22 August 2017 to debug some erroneous behavior that they witnessed during internal testing.  It looks like all of these server errors boil down to Hibernate throwing a GeneralError.  This ticket will track the diagnosis and fix for these problems.

**Comment by kiniry** [comment]
There are nearly 1,000 `GeneralError` messages in the server log.  Most relate to a `NullPointerException` being thrown inside of Hibernate.  The (truncated) call traces that trigger NPEs are:
1. ```us.freeandfair.corla.persistence.Persistence.commitTransaction(Persistence.java:330)
us.freeandfair.corla.endpoint.AbstractEndpoint.after(AbstractEndpoint.java:360)```
2. ```us.freeandfair.corla.query.ElectorQueries.matching(ElectorQueries.java:75)
us.freeandfair.corla.endpoint.EstablishAuditBoard.endpoint(EstablishAuditBoard.java:91)```
3. us.freeandfair.corla.persistence.Persistence.getAll(Persistence.java:542)
us.freeandfair.corla.json.DoSDashboardRefreshResponse.countyStatusMap(DoSDashboardRefreshResponse.java:162)
us.freeandfair.corla.json.DoSDashboardRefreshResponse.createResponse(DoSDashboardRefreshResponse.java:118)
us.freeandfair.corla.endpoint.DoSDashboardRefresh.endpoint(DoSDashboardRefresh.java:60)```
4. ```us.freeandfair.corla.persistence.Persistence.getAll(Persistence.java:542)
us.freeandfair.corla.endpoint.ContestDownload.endpoint(ContestDownload.java:68)```
5. ```us.freeandfair.corla.query.CountyQueries.forAdministrator(CountyQueries.java:70)
us.freeandfair.corla.endpoint.Authentication.authenticatedCounty(Authentication.java:215)
us.freeandfair.corla.endpoint.CountyDashboardRefresh.endpoint(CountyDashboardRefresh.java:59)```

**Comment by kiniry** [comment]
On occasion, we are also seeing an `UnsupportedOperationException` with the following call trace:
```
Aug 22 15:29:34 rh7rlapoc faf_server.sh: 15:29:34,536 ERROR GeneralError:54 -
Aug 22 15:29:34 rh7rlapoc faf_server.sh: java.lang.UnsupportedOperationException: Attempted to serialize java.lang.Class: org.hibernate.proxy.HibernateProxy. Forgot to register a type adapter?
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.TypeAdapters$1.write(TypeAdapters.java:73)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.TypeAdapters$1.write(TypeAdapters.java:69)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.TypeAdapter$1.write(TypeAdapter.java:191)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:69)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.ArrayTypeAdapter.write(ArrayTypeAdapter.java:95)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:69)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:125)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:243)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:69)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:125)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:243)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.ObjectTypeAdapter.write(ObjectTypeAdapter.java:107)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:69)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:97)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:61)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.Gson.toJson(Gson.java:669)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.Gson.toJson(Gson.java:648)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at com.google.gson.Gson.toJson(Gson.java:623)
Aug 22 15:29:34 rh7rlapoc faf_server.sh: at us.freeandfair.corla.endpoint.ContestDownloadByCounty.endpoint(ContestDownloadByCounty.java:80)
```

**Comment by dmzimmerman** [comment]
The `NullPointerException`, I'm reasonably sure, is caused by a database problem or (less likely) a Hibernate bug. I find myself wondering if we dropped and recreated their database from scratch when we installed the latest version of the server, because if we didn't, it's possible that there were some leftover records somewhere that had no version fields (hence causing the NPE). I cannot think of another reason it might occur; we have never seen it in any of our testing, and the version fields are strictly read-only in our code and are never set to null by us. If it's a Hibernate bug, we'll keep seeing it; if not, it should be solved by dropping and recreating the database.

The `UnsupportedOperationException`, on the other hand, is due to lazy loading, and can be solved by actually loading the objects as we serialize them. I'll fix those.
